library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Ivoz\Provider\Application\Service\Carrier;
+
+use Assert\Assertion;
+use Ivoz\Core\Application\DataTransferObjectInterface;
+use Ivoz\Core\Application\Service\Assembler\CustomDtoAssemblerInterface;
+use Ivoz\Core\Domain\Model\EntityInterface;
+use Ivoz\Provider\Application\Service\CarrierServer\StatusFromLcrGatewayInfo;
+use Ivoz\Provider\Domain\Model\Carrier\CarrierDto;
+use Ivoz\Provider\Domain\Model\Carrier\CarrierInterface;
+use Ivoz\Provider\Domain\Model\CarrierServer\CarrierServer;
+use Ivoz\Provider\Domain\Model\CarrierServer\CarrierServerRepository;
+use Ivoz\Provider\Domain\Model\Carrier\CarrierStatus;
+
+class CarrierDtoAssembler implements CustomDtoAssemblerInterface
+{
+    public function __construct(
+        private CarrierServerRepository $carrierServerRepository,
+        private StatusFromLcrGatewayInfo $statusFromLcrGatewayInfo,
+    ) {
+    }
+
+    public function toDto(
+        EntityInterface $entity,
+        int $depth = 0,
+        string $context = null
+    ): DataTransferObjectInterface {
+        Assertion::isInstanceOf(
+            $entity,
+            CarrierInterface::class
+        );
+
+        $carrierDto = $entity->toDto($depth);
+
+        $carrierServers = $this
+            ->carrierServerRepository
+            ->findByCarrierId(
+                (int) $entity->getId()
+            );
+
+        $carrierDto->addStatus(
+            new CarrierStatus(
+                true
```

Comment:
Is this suppoused to be true when no carrier servers are found?

Reply:
No, sorry, this is a logic error.